ibis/expr/analysis.py

```diff
@@ -541,54 +541,19 @@ def get_mutation_exprs(
     # 1) overwriting_cols_to_expr, which maps a column name to its expr
     # if the expr contains a column that overwrites an existing table column.
     # All keys in this dict are columns in the original table that are being
-    # overwritten by an assignment expr. All values in this dict are either:
-    #     (a) The expr of the overwriting column. Note that in the case of
-    #     DestructColumn, this will specifically only happen for the first
-    #     overwriting column within that expr.
-    #     (b) None. This is the case for the second (and beyond) overwriting
-    #     column(s) inside the DestructColumn and is used as a flag to prevent
-    #     the same DestructColumn expr from being duplicated in the output.
+    # overwritten by an assignment expr.
     # 2) non_overwriting_exprs, which is a list of all exprs that do not do
     # any overwriting. That is, if an expr is in this list, then its column
     # name does not exist in the original table.
     # Given these two data structures, we can compute the mutation node exprs
     # based on whether any columns are being overwritten.
     # TODO issue #2649
-    # Due to a known limitation with how we treat DestructColumn
-    # in assignments, the ordering of op.selections may not exactly
-    # correspond with the column ordering we want (i.e. all new columns
-    # should appear at the end, but currently they are materialized
-    # directly after those overwritten columns).
     overwriting_cols_to_expr: dict[str, ir.Expr | None] = {}
     non_overwriting_exprs: list[ir.Expr] = []
     table_schema = table.schema()
     for expr in exprs:
-        is_first_overwrite = True
         expr_contains_overwrite = False
-        if isinstance(expr, ir.DestructColumn):
```

Comment:
Do the new code produces the same column ordering as before? I remembered this was a pretty nasty issue before.

Reply:
(When part of the destructure columns overwrites existing columns)

Reply:
The column ordering now should be identical as calling

```python
t.mutate(
    t.struct_field['a'],
    t.struct_field['b'],
    t.struct_field['c'],
)
```

Reply:
kk sounds good. Just double checking.